designer/server/src/main/scala/pl/touk/nussknacker/ui/process/migrate/RemoteEnvironment.scala

```diff
@@ -112,17 +116,28 @@ trait StandardRemoteEnvironment extends FailFastCirceSupport with RemoteEnvironm
       _ <- processEither match {
         case Right(processDetails) if processDetails.isArchived => EitherT.leftT[Future, EspError](MigrationToArchivedError(processDetails.idWithName.name, environmentId))
         case Right(_) => EitherT.rightT[Future, EspError](())
-        case Left(RemoteEnvironmentCommunicationError(StatusCodes.NotFound, _)) => createProcessOnRemote(localProcess, category)
+        case Left(RemoteEnvironmentCommunicationError(StatusCodes.NotFound, _)) =>
+          if(passUsername) createScenarioOnRemote(localProcess, category, loggedUser) else createProcessOnRemote(localProcess, category)
         case Left(other) => EitherT.leftT[Future, EspError](other)
       }
-      _ <- EitherT.right[EspError](saveProcess(localProcess, UpdateProcessComment(s"Scenario migrated from $environmentId by ${loggedUser.username}")))
+      usernameToPass = if(passUsername) Some(RemoteUserName(loggedUser.username)) else None
+      _ <- EitherT { saveProcess(localProcess, UpdateProcessComment(s"Scenario migrated from $environmentId by ${loggedUser.username}"), usernameToPass) }
     } yield ()).value
   }
 
+  private def createScenarioOnRemote(localProcess: DisplayableProcess, category: String, loggedUser: LoggedUser)
```

Comment:
Let's have a one method (`createScenarioOnRemote`) instead of two with loggedUser as an Option

Reply:
When we removed `migrateScenario` it was unnecessary to have it - removed. :+1: